src/Servers/Kestrel/Core/src/Internal/Infrastructure/TransportManager.cs

```diff
@@ -179,7 +178,14 @@ public override CancellationToken ConnectionClosed
 
     public Task StopEndpointsAsync(List<EndpointConfig> endpointsToStop, CancellationToken cancellationToken)
     {
-        var transportsToStop = _transports.Where(t => t.EndpointConfig != null && endpointsToStop.Contains(t.EndpointConfig)).ToList();
+        var transportsToStop = new List<ActiveTransport>();
```

Comment:
How many do we expect? Almost none? Almost all? I'm thinking "list initial size" reasons

Reply:
At most it is `_transports.Count`.  At minimum, it is `0`. I'm not sure what the typical case would be.

This is only called on `RebindAsync`, which happens when the configuration changes and there are endpoints to stop.

Reply:
I'd leave it. There are much bigger optimizations we could make to endpoint rebinding if we cared more about this scenario. For example, we could start bringing new endpoints online before completely stopping all the removed ones if they don't conflict.